### ISSUE TITLE
Changing build action to "Page" causes the file to be disappear from solution explorer and can never be included

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#3028]
1. File -> New Console App (.NET COre)
2. Change build action of Program.cs -> Page

-- Notice it disappears --

If you show all files it shows but you cannot include it.

**Comment by Pilchie** [comment]
Any idea why this happens?  Do we have metadata to hide Page items somewhere?

**Comment by davkean** [comment]
Normally this would be because we haven't marked the rule ("Page") as being part of the File context, but it looks like we are:

``` XML
    <PropertyPageSchema Include="$(ManagedXamlResourcesDirectory)Page.xaml">
      <Context>File;BrowseObject</Context>
    </PropertyPageSchema>
```

I'm not sure if this our bug or CPS.

**Comment by Pilchie** [comment]
@jviau @lifengl - any ideas?

**Comment by jviau** [comment]
CPS is being told to look for Page.xaml in `{VisualStudioInstallDir}\MSBuild\Microsoft\VisualStudio\Managed`, but it is not there.

**Comment by lifengl** [comment]
Can this be the same problem that a page item can not be found in the tree in shared projects?

Sent from my phone

On Jan 10, 2018, at 2:08 PM, Jacob Viau <notifications@github.com<mailto:notifications@github.com>> wrote:


CPS is being told to look for Page.xaml in {VisualStudioInstallDir}\MSBuild\Microsoft\VisualStudio\Managed, but it is not there.

**Comment by jviau** [comment]
No, this only affects .netcore projects. That bug you mentioned repros in shared projects.

**Comment by jviau** [comment]
Further investigation reveals this is related. Adding xaml files to shared projects only works if you have had the UWP workload installed, because that adds a Page.xaml. Without UWP, shared projects will look for Page.xaml in the location above and fail to find it. Causing our page item not found NFE.

**Comment by Pilchie** [comment]
Pulling into 15.7 to look at for Xamarin forms scenarios.

**Comment by pierceboggan** [comment]
@Pilchie Is this still slated for 15.7?

**Comment by Pilchie** [comment]
@jviau do you know where the `Page.xaml` in UWP comes from?  Maybe we should just move it here?

**Comment by Pilchie** [comment]
@pierceboggan - we'll still try.

**Comment by jviau** [comment]
The page.xaml from UWP is in `Enterprise\MSBuild\Microsoft\WindowsXaml\v15.0`. It is installed via the UWP workload. I don't know if removing it from there will work, then UWP may only work if .netcore workload is also installed.

**Comment by Pilchie** [comment]
I'm pretty sure everything from this repo except templates is installed in the UWP workload in order to support .NET Standard projects.  @jmarolf can probably confirm.

**Comment by pierceboggan** [comment]
@Pilchie Did this fix end up making it into 15.6?

**Comment by Pilchie** [comment]
Not yet, and at this point it's unlikely to fit :-/

**Comment by davkean** [comment]
Have a fix for this.
